### DISCUSSION TITLE
Allowed memory size on Synology

### DISCUSSION THREAD

**Discussion by Seruga123** [kimai/kimai#3468] [On-Premise: Installation & Updates]
Hi

I have been trying to install Kimai2 on my Synology NAS for weeks. With these instructions https://www.kimai.org/static/downloads/Installationsanleitung_Synology_DSM7_Kimai.pdf I can install and use version 1.17.1 without any problems. But no matter if I want to update to version 1.22.1 or install this version from the beginning, I always get the error message "Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 32768 bytes)". Of course I have also rebuilt the cache as described here https://www.kimai.org/documentation/cache.html and checked the permissions. https://www.kimai.org/documentation/synology.html (I have tried the permissions in different variants, also with full rights to all folders and also directly via Synology File Station, all without success).

I have already tried a lot of installation variants, but I just can't get any further. Maybe someone could create a new step-by-step guide like this https://www.kimai.org/static/downloads/Installationsanleitung_Synology_DSM7_Kimai.pdf, but not for version 1.17.1 but for version 1.22.1? 

I would be very grateful

Translated with www.DeepL.com/Translator (free version)

**Comment by kevinpapst** [comment]
You have to increase your `memory_limit` of your Synology PHP installation.

I don't know whether this has to be done in the `php.ini` or through the WebUI, I am not a Sinology user.

**Reply by BeckeBauer** [reply]
@Seruga123
You have to SSH into your Syno NAS in order to increase the php memory. There are many tutorials on the internet in case you have never done so.
I assume that you have installed the php7.4 package and use it with kimai. In this case, you find the php.ini here:
`/usr/local/etc/php74/cli` 
Navigate to this folder and open the php.ini
Find the line `memory_limit = xxM` and increase xx to an higher value and save the file; I am using 512M for example
Run the kimai installation again
Please keep in mind that the changes will be overwritten after a Syno NAS update so you will have to do it again.

**Reply by Seruga123** [reply]
Many, many thanks BeckeBauer

That was indeed the solution. Even though I don't understand it at all. As described in the instructions, I deliberately created a new PHP script via Synology WebStation for Kimai2 only and entered 512M under Core - memory_limit. That's why I assumed that it couldn't have anything to do with this. It is incomprehensible to me that this setting does not apply to the file via ssh /usr/local/etc/php74/cli/php.ini. I even increased the memory_limit in the Synology default PHP 7.4 profile to 512M and this apparently has no effect on php.ini. I do not understand this at all. Why can I create such scripts at all if the settings do not take effect? I'll have to ask Synology about that. 

But long story short: You have helped me a lot. I have already invested many days of time and it has never worked. Now the installation works perfectly. Many, many thanks. 

Translated with www.DeepL.com/Translator (free version)

**Reply by BeckeBauer** [reply]
👍 I am happy that it worked... Struggled with the same issues when I started to used synology. I assume that the php profile in Webstation which you have configured to run Kimai is not used when you install Kimai from command line...

**Reply by Seruga123** [reply]
Oh, I see, that makes sense. Thank you for the clarification. 

@kevinpapst Another question at this point, perhaps directly to the Kimai developers. Is there already or perhaps in the future an installation variant as known from other open source projects? For example, copy the files from a ZIP directly to any web server, assign permissions, create a database and then run an installation wizard via https://exempel.com/install.php and configure it. The link to the database, the creation of the tables, etc. should also be possible in this way. This way everything would not have to be done via ssh and I think the installation would be simplified? 

Translated with www.DeepL.com/Translator (free version)

**Reply by kevinpapst** [reply]
I adjusted the Synology documentation and added a point to the FAQ. Thanks @BeckeBauer 👍
